### ISSUE TITLE
Enhancement: include file names in search results

### ISSUE THREAD

**Issue by lemongrassnginger** [NaturalDocs/NaturalDocs#6]
First off - really impressed with this tool! Love the ease of commenting and the navigation and layout of the output :)

The search feature is great and becomes pretty important in large projects. While searching for section titles in [extra documentation](http://www.naturaldocs.org/getting_started/documenting_your_code/#extra_documentation) is supported, I don't think searching by file name is - something that I think readers will naturally expect to be able to do (I certainly find myself searching for files by name!). A work-around is to include the name of the file in the section title of the document. but it'd be nice for this to just work straight out. 

Adding/modifying the relevant prefix file in `search/keywords/` as follows seems to work (to add the README.txt, for example): 

```
NDSearch.OnPrefixDataLoaded("rea", ["Section"], [
	["README", , [
		[, , , , 1, "File:README.txt"]
	]]
]);
```

If anyone's able to give some pointers as where to start I'd be happy to give this a go and submit a PR. I've mostly done JS development in the past.

**Comment by NaturalDocs** [comment]
Makes sense.  I'll add this to my to do list.  It wouldn't be the JavaScript so much as the C# code that generates the JavaScript indexes.  I can take care of that.
